tests/test_windowing.py

```diff
@@ -0,0 +1,310 @@
+            DataFrame,
+        ),
+        DataFrame,
+    )
+
+    # func: np.ufunc | Callable | str | list[Callable | str, np.ufunc] | dict[Hashable, Callable | str | np.ufunc| list[Callable | str]]
+    check(assert_type(DF.rolling(10).agg("sum"), DataFrame), DataFrame)
+
+
+def test_rolling_basic_math_series() -> None:
+    check(assert_type(S.rolling(10, min_periods=10).count(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).sum(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).mean(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).median(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).var(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).std(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).min(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).max(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).corr(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).cov(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).skew(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).kurt(), Series), Series)
+    check(assert_type(S.rolling(10, min_periods=10).sem(), Series), Series)
+    check(assert_type(S.rolling(10).quantile(0.5), Series), Series)
+    check(assert_type(S.rolling(10).rank("average"), Series), Series)
+    check(assert_type(S.rolling(10).rank("min"), Series), Series)
+    check(assert_type(S.rolling(10).rank("max"), Series), Series)
+
+
+def test_rolling_apply_series() -> None:
+    check(assert_type(S.rolling(10).apply(np.mean), Series), Series)
+
+    def _mean(df: Series) -> float:
+        return df.mean()
+
+    check(assert_type(S.rolling(10).apply(_mean), Series), Series)
+
+    def _mean2(df: Series) -> np.ndarray:
+        return np.mean(df, axis=0)
+
+    check(assert_type(S.rolling(10).apply(_mean2, raw=True), Series), Series)
+
+
+def test_rolling_aggregate_series() -> None:
+    check(assert_type(S.rolling(10).aggregate(np.mean), Series), Series)
+    check(assert_type(S.rolling(10).aggregate("mean"), Series), Series)
+
+    def _mean(s: Series) -> float:
+        return s.mean()
+
+    check(assert_type(S.rolling(10).aggregate(_mean), Series), Series)
+
+    check(assert_type(S.rolling(10).aggregate([np.mean]), DataFrame), DataFrame)
+    check(assert_type(S.rolling(10).aggregate([np.mean, "mean"]), DataFrame), DataFrame)
+    check(
+        assert_type(
+            S.rolling(10).aggregate({"col1": np.mean, "col2": "mean", "col3": _mean}),
+            DataFrame,
+        ),
+        DataFrame,
+    )
+    # check(assert_type(S.rolling(10).aggregate({"col1": [np.mean, "mean"], "col2": "mean"}), Series), DataFrame)
+
+    # func: np.ufunc | Callable | str | list[Callable | str, np.ufunc] | dict[Hashable, Callable | str | np.ufunc| list[Callable | str]]
```

Comment:
why is this commented out??

Reply:
It is invalid code.  Was just part of setting up tests.